src/gstvimbasrc.c

```diff
@@ -1558,6 +1559,10 @@ VmbError_t set_roi(GstVimbaSrc *vimbasrc)
                            ErrorCodeToMessage(result));
     }
     // offsetx
+    if (vimbasrc->properties.offsetx == INT_MAX) {
+        VmbInt64_t vmb_offsetx = (vmb_width - vimbasrc->properties.width) >> 1;
```

Comment:
Add a debug message so we can easier understand why this code path was chosen

```suggestion
        VmbInt64_t vmb_offsetx = (vmb_width - vimbasrc->properties.width) >> 1;
        GST_DEBUG_OBJECT(vimbasrc, "ROI centering along x-axis requested. Calculated offsetx=%d",
                         vmb_offsetx);
```